src/misc.jl

```diff
@@ -404,7 +360,7 @@ discretize_make_ia(values::CategoricalArray) =
     discretize_make_ia(values, intersect(push!(levels(values), missing), unique(values)))
 discretize_make_ia(values::CategoricalArray, ::Nothing) = discretize_make_ia(values)
 function discretize_make_ia(values::CategoricalArray{T}, levels::Vector) where {T}
-    mapping = Union{Nothing,Int}[coalesce.(indexin(CategoricalArrays.index(values.pool), levels), 0)...]
+    mapping = something(indexin(CategoricalArrays.index(values.pool), levels), 0)
```

Comment:
I think this should be `something.(...)`. Also, below `coalesce` should be replaced with `something`. It's weird that tests didn't catch it.